toolchain/lower/handle_aggregates.cpp

```diff
@@ -118,12 +118,31 @@ static auto EmitAggregateInitializer(FunctionContext& context,
                                      SemIR::InstBlockId refs_id,
                                      llvm::Twine name) -> llvm::Value* {
   auto* llvm_type = context.GetType(type_id);
+  auto refs = context.sem_ir().inst_blocks().Get(refs_id);
 
   switch (SemIR::InitRepr::ForType(context.sem_ir(), type_id).kind) {
-    case SemIR::InitRepr::None:
-    case SemIR::InitRepr::InPlace:
+    case SemIR::InitRepr::None: {
       // TODO: Add a helper to poison a value slot.
       return llvm::PoisonValue::get(llvm_type);
+    }
+
+    case SemIR::InitRepr::InPlace: {
+      // Finish initialization of constant fields. We will have skipped this
+      // when emitting the initializers because they have constant values.
```

Comment:
Can we add a TODO to revisit the order of in which we emit these? As discussed live, the current order of evaluation and emission results in I think a somewhat surprising / unfortunate order of effects between storing constants and evaluating non-constant initialization.

I understand this will be quite complicated, and clearly the current PR is an improvement, but I want to leave some note that we should revisit this and think about exactly how we want this to work.

Reply:
TODO added.